workers/unity/Packages/io.improbable.gdk.tools/CodegenLogUtils.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using UnityEngine;
+
+namespace Improbable.Gdk.Tools
+{
+    internal static class CodegenLogUtils
+    {
+        internal enum CodegenLogLevel
+        {
+            TRACE,
+            DEBUG,
+            INFO,
+            WARN,
+            ERROR,
+            FATAL,
+            OFF
+        }
+
+        internal struct CodegenLog
+        {
+            public readonly string time;
+            public readonly CodegenLogLevel level;
+            public readonly string logger;
+            public readonly string message;
+            public readonly string exception;
+            public bool isException => !string.IsNullOrEmpty(exception);
+
+            [Serializable]
+            private struct RawCodegenLog
+            {
+#pragma warning disable 649
+                public string time;
+                public string level;
+                public string logger;
+                public string message;
+                public string exception;
+#pragma warning restore
+            }
+
+            public CodegenLog(string rawLog)
```

Comment:
Prefer a `public static CodegenLog FromRaw(string rawLog)`

Makes the intention clearer

Reply:
Also should this be private if its only called from `ProcessCodegenLogs` ?